### PR TITLE
Update FlexAttention with masking semantic

### PR DESCRIPTION
**Pull request by drisspg**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133373

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @mikaylagawarecki

### CODE REVIEW

File: test/inductor/test_flex_attention.py [hunk 1]
```diff
@@ -212,7 +213,7 @@ def _check_equal(
         compiled_error = (golden_out - compiled_out).abs().mean()
         ref_error = (golden_out - ref_out).abs().mean()
         # TODO: Make this check stricter after updating eager SDPA masked_softmax semantics
-        if torch.isnan(compiled_error).any() and not torch.isnan(ref_error).any():
+        if torch.isnan(compiled_error).any() or torch.isnan(ref_error).any():
```
Comment (yanboliang) [inline test/inductor/test_flex_attention.py:216]: Probably remove the comment above.

**Review by yanboliang — APPROVED** [review]

**Review by Chillee — COMMENTED** [review]
Can you add a test like this?
```
out, lse = torch.compile(flex_attention)(q, k, v, score_mod=mask, return_lse=True)
lse.sum().backward()
```

**Pull request closed without merging**